.github/workflows/tests.yml

```diff
@@ -1,43 +1,49 @@
-name: tests
+name: Tests
 
 on:
-  push: null
-  pull_request: null
+  pull_request:
+  push:
+    branches: [master]
 
 jobs:
-  tests:
-    name: tests
-    runs-on: "ubuntu-latest"
+  run:
+    runs-on: ${{ matrix.os }}
+    strategy:
+      matrix:
+        python-version: ["3.6", "3.7", "3.8", "3.9", "3.10"]
+        os: [windows-latest, ubuntu-latest, macos-latest]
+      fail-fast: false
+
     steps:
       - name: cancel previous runs
-        uses: styfle/cancel-workflow-action@0.6.0
+        uses: styfle/cancel-workflow-action@0.10.0
         with:
           access_token: ${{ github.token }}
 
-      - uses: actions/checkout@v2
+      - uses: actions/checkout@v3
 
-      - uses: conda-incubator/setup-miniconda@v2
+      - name: Setup Micromamba
+        uses: mamba-org/provision-with-micromamba@main
         with:
-          python-version: 3.8
-          channels: conda-forge,defaults
-          channel-priority: strict
-          show-channel-urls: true
-
-      - name: configure conda and install code
+         environment-file: false
+ 
+      - name: Python ${{ matrix.python-version }}
         shell: bash -l {0}
         run: |
-          conda config --set always_yes yes
-          conda config --add channels conda-forge
-          conda install --quiet pip
-
-          export GIT_FULL_HASH=`git rev-parse HEAD`
-          export CICLE_BUILD_URL="https://www.youtube.com/watch?v=R7qT-C-0ajI"
-          conda install --file requirements.txt --file requirements-dev.txt
+          micromamba create --name TEST python=${{ matrix.python-version }} pip --file requirements-dev.txt --channel conda-forge
```

Comment:
Could add `python` and `pip` in [extra-specs](https://github.com/mamba-org/provision-with-micromamba#extra-specs) above and then just update the environment here instead of create. And then maybe the activate lines wouldn't be needed?

Reply:
I find that a bit confusing b/c I like the CIs to be as close as what a user would do locally.

Reply:
That is understandable. Why not just use setup-python instead then? I think provision-with-micromamba is great, but if you don't even have a conda env, then maybe it is overkill.

Reply:
For this project it would not be that different, you are correct. But I only use conda packages so this makes it easier for me to port stuff from one project to another.

Reply:
That's fair!